Pristine-1.4 error while compiling kernel

Hi, I got this error when compiling the the kernel (commit 459cc46) and I get the following error:

```
$ sudo make bzImage modules
  CHK     include/config/kernel.release
  CHK     include/generated/uapi/linux/version.h
  CHK     include/generated/utsrelease.h
  CHK     include/generated/bounds.h
  CHK     include/generated/asm-offsets.h
  CALL    scripts/checksyscalls.sh
  CHK     include/generated/compile.h
  CC      net/rina/ipcp-factories.o
In file included from include/linux/linkage.h:4:0,
                 from include/linux/kernel.h:6,
                 from net/rina/logs.h:28,
                 from net/rina/ipcp-factories.c:23:
net/rina/ipcp-factories.c: In function ‘ipcpf_fini’:
net/rina/ipcp-factories.c:83:42: error: ‘struct rset’ has no member named ‘list’
         ASSERT(list_empty(&factories->set->list));
                                          ^
include/linux/compiler.h:164:42: note: in definition of macro ‘unlikely’
 # define unlikely(x) __builtin_expect(!!(x), 0)
                                          ^
net/rina/debug.h:28:22: note: in expansion of macro ‘BUG_ON’
 #define ASSERT(COND) BUG_ON(!(COND))
                      ^
net/rina/ipcp-factories.c:83:9: note: in expansion of macro ‘ASSERT’
         ASSERT(list_empty(&factories->set->list));
         ^
scripts/Makefile.build:258: recipe for target 'net/rina/ipcp-factories.o' failed
make[2]: *** [net/rina/ipcp-factories.o] Error 1
scripts/Makefile.build:403: recipe for target 'net/rina' failed
make[1]: *** [net/rina] Error 2
Makefile:947: recipe for target 'net' failed
make: *** [net] Error 2
```

Commenting the line (`net/rina/ipcp-factories.c:83`) did solve the problem for me but it may be good to fix the assert to avoid future errors?

--- a/linux/net/rina/ipcp-factories.c
+++ b/linux/net/rina/ipcp-factories.c
@@ -79,8 +79,12 @@ int ipcpf_fini(struct ipcp_factories * factories)
                 return -1;
         }
 
-        /* All thetemplates have to be unregistered from now on */
-        ASSERT(list_empty(&factories->set->list));
+        /* All the templates have to be unregistered from now on */
+	/* In order to maintain the robject abstraction the kset should not be
+	 * accessed here. If accesing to the internal list is required
+	 * somewhere else, an API must be added. Since this is debugging code we
+	 * can use this workaround in the meantime */
+        ASSERT(list_empty(&factories->set->kset->list));
 
         rset_unregister(factories->set);
 
@@ -103,13 +107,8 @@ struct ipcp_factory * ipcpf_find(struct ipcp_factories * factories,
                 return NULL;
 
         k = rset_find_obj(factories->set, name);
-        if (k) {
-		/* kobject_put is not needed since the struct factory is the one to be
-		 * freed
-                robject_put(k);
-		*/
+        if (k)
                 return to_ipcp(k);
-        }
 
         return NULL;
 }

